app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/shops/ShopsScreen.kt

```diff
@@ -46,7 +47,11 @@ import org.koin.androidx.compose.getViewModel
 
 @OptIn(ExperimentalMaterialApi::class, FlowPreview::class)
 @Composable
-fun ShopsScreen(shopsViewModel: ShopsViewModel) {
+fun ShopsScreen(
+    shopsViewModel: ShopsViewModel,
+    shopDetailsViewModel: ShopDetailsViewModel,
+    navToShopDetails: () -> Unit = {},
```

Comment:
Rename this to onShopClicked which will take shop as an argument. Then in the ShopsMainScreen you can do the `shopDetailsViewModel.setLastClickedShop(shop)` and there will be no need to pass down shopDetailsViewModel.